### PR TITLE
[graphql/rpc] stake comments for docs

### PR DESCRIPTION
**Pull request by stefan-mysten**
## Description 

Adds comments for the Stake fields and stake connection.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by amnn — COMMENTED** [review]
Currently these docs aren't revealing much more than the name of the field/function they are attached to.  Can we expand them a bit? E.g.

- `stake_connection`: Explain that `stake` corresponds to `StakedSui` objects owned by the address/object.
- Explain what it means for a `StakedSui` to be active -- I believe it is the epoch at which that stake starts earning rewards, but double check that.
- Estimated Reward: Explain how that's calculated.
- Principal: Describe this without using the word "principal" (This is a good "lint" for doc comments in general).
- Status: What does each status mean? (This documentation probably belongs on the `StakeStatus` type, so feel free to ignore if it's already there).

File: crates/sui-graphql-rpc/src/types/stake.rs [hunk 1]
```diff
@@ -17,20 +17,27 @@ pub(crate) enum StakeStatus {
 #[derive(Clone, PartialEq, Eq, SimpleObject)]
 #[graphql(complex)]
 pub(crate) struct Stake {
+    /// Stake object address
     pub id: ID,
+    /// The epoch at which the stake became actives
```
Comment (ronny-mysten) [inline crates/sui-graphql-rpc/src/types/stake.rs:22]:
```suggestion
    /// The epoch at which the stake became active
```

**Review by ronny-mysten — APPROVED** [review]

File: crates/sui-graphql-rpc/src/types/stake.rs [hunk 2]
```diff
@@ -56,6 +67,8 @@ impl Stake {
             Ok(None)
         }
     }
+
+    /// The Stake object as a Move object
```
Comment (amnn) [inline crates/sui-graphql-rpc/src/types/stake.rs:71]: nit: Worth drawing attention to the corresponding Move type here.

```suggestion
    /// The corresponding StakedSui Move object
```

**Review by amnn — APPROVED** [review]
Much clearer, thank you!

**Pull request merged**